frontend/src/components/QueryBuilder/QueryBuilder.tsx

```diff
@@ -2036,69 +1809,48 @@ function QueryBuilder(props: QueryBuilderProps) {
 				m={readonly ? undefined : '8'}
 				shadow="medium"
 			>
-				{mode !== QueryBuilderMode.EMPTY && (
-					<Box
-						p="4"
-						paddingBottom="8"
-						background="white"
-						borderBottom={readonly ? undefined : 'secondary'}
-						display="flex"
-						alignItems="center"
-						flexWrap="wrap"
-						gap="4"
-					>
-						{filterRules.flatMap((rule, index) => [
-							...(index != 0
-								? [
-										<Tag
-											shape="basic"
-											kind="secondary"
-											emphasis="low"
-											onClick={toggleIsAnd}
-											key={`separator-${index}`}
-											disabled={readonly}
-										>
-											{isAnd ? 'and' : 'or'}
-										</Tag>,
-								  ]
-								: []),
-							<QueryRule
-								key={`rule-${index}`}
-								rule={rule}
-								onChangeKey={(val) => {
-									// Default to 'is' when rule is not defined yet
-									if (rule.op === undefined) {
-										updateRule(rule, {
-											field: val,
-											op: getDefaultOperator(rule.field),
-										})
-									} else {
-										updateRule(rule, { field: val })
-									}
-								}}
-								getKeyOptions={getKeyOptions}
-								onChangeOperator={(val) => {
-									if (val?.kind === 'single') {
-										updateRule(rule, { op: val.value })
-									}
-								}}
-								getOperatorOptions={getOperatorOptionsCallback(
-									getCustomFieldOptions(rule.field),
-									rule.val,
-								)}
-								onChangeValue={(val) => {
-									updateRule(rule, { val: val })
-								}}
-								getValueOptions={getValueOptionsCallback(
-									rule.field,
-								)}
-								onRemove={() => removeRule(rule)}
-								readonly={readonly ?? false}
-							/>,
-						])}
-						{addFilterButton}
-					</Box>
-				)}
+				<Box
+					p="4"
+					paddingBottom="8"
+					background="white"
+					borderBottom={readonly ? undefined : 'secondary'}
+					display="flex"
+					alignItems="center"
+					flexWrap="wrap"
+					gap="4"
+				>
+					{filterRules.flatMap((rule, index) => [
+						...(index != 0
+							? [
+									<Tag
+										shape="basic"
+										kind="secondary"
+										emphasis="low"
+										onClick={toggleIsAnd}
+										key={`separator-${index}`}
+										disabled={readonly}
+									>
+										{isAnd ? 'and' : 'or'}
```

Comment:
@mayberryzane i think there is a new bug that causes results to not update when switching `or/and`
https://www.loom.com/share/daac0ac7f5a04c4fbc168cccf5286835

Reply:
same bug exists in prod rn but it's probably an easy fix, taking a look

Reply:
sg, if u'd prefer to fix in separate pr, feel free to ship this